pkgs/development/libraries/gperftools/default.nix

```diff
@@ -51,6 +51,26 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  # split output to minimize dependencies on dynamic linking.
+  # tcmalloc_minimal goes into a dedicated output to further minimize
+  # the runtime dependencies of applications that just use tcmalloc
+  # as an allocator, without the profiling.
+  outputs = [ "out" "dev" "tcmalloc_minimal" "lib" "static" ];
```

Comment:
I'm not sure splitting `dev` and `static` provides additional value over putting archives into `dev`.

Reply:
It does. When NIX_LDFLAGS gets constructed in the stdenv hooks, there's a particular fixed order, where `dev` goes first in the list, before any of the propagated build outputs. This means that if the static archives are in `dev`, doing something like
```
gcc -o foo foo.c -ltcmalloc
```
will attempt to link against the static library.

Reply:
I don't quite understand where this order is coming from, but I confirmed this behavior experimentally. I suppose it is one of the unintentional behaviors of `stdenv`.

Reply:
The ordering is happening here: https://github.com/NixOS/nixpkgs/blob/3a3585194796fd431880776dbe313252d1070378/pkgs/build-support/setup-hooks/multiple-outputs.sh#L183
The `dev` output is set up to propagate the other propagatedBuildOutputs. That fixes the order of `dev` relative to the others.